gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -75,15 +75,27 @@ object TransformHints {
   }
 
   def tag(plan: SparkPlan, hint: TransformHint): Unit = {
-    if (isAlreadyTagged(plan)) {
-      if (isNotTransformable(plan) && hint.isInstanceOf[TRANSFORM_SUPPORTED]) {
-        throw new UnsupportedOperationException(
-          s"Plan was already tagged as non-transformable, " +
-            s"cannot mark it as transformable after that: ${plan.toString()}")
+    val mergedHint = getHintOption(plan)
+      .map {
+        case TRANSFORM_UNSUPPORTED(Some(originalReason)) =>
+          hint match {
+            case TRANSFORM_UNSUPPORTED(Some(newReason)) =>
+              untag(plan)
+              TRANSFORM_UNSUPPORTED(Some(originalReason + "; " + newReason))
+            case TRANSFORM_UNSUPPORTED(None) =>
+              untag(plan)
+              hint
```

Comment:
Here, we should use original hint since new `hint` has no reason, right?

Reply:
yeah, we should use original hint here